### ISSUE TITLE
Editing `Date picker` fields via XB is broken

### ISSUE THREAD

**Issue by phuongnq** [craftercms/craftercms#6550]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the develop branch
- [X] The issue is in the latest released 4.1.x
- [X] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

CI: 4768

When opening a date picker UI element, it appears positioned off the top of the frame. On large windows, you can scroll back up to see the rest of it, but in XB, scrolling doesn't work, so the item is not editable.

### Steps to reproduce

Steps:

1. Create a site from Editorial BP
2. Update Home content type to have a DateTime control with name `date_dt`
3. Update Home content to have a value for `date_dt`
4. Update home.ftl template to display the `date_dt`:

```
<@crafter.time $field="date_dt">
        ${model.date_dt?datetime}
      </@crafter.time>
```
5. While in preview mode with XB enabled, click to edit the `date_dt`
6. On opening popup, click to update the data
7. Notice that the datetime picker is cut-off in some case and cannot edit

### Screenshots and/or videos

![2024-02-16_16-41](https://github.com/craftercms/craftercms/assets/2996543/aa451976-e400-4f15-97a3-999eca56af3a)
![2024-02-16_16-23](https://github.com/craftercms/craftercms/assets/2996543/0dedbbb0-603a-4bbf-9d95-9b25f0073fd0)

**Comment by rart** [comment]
@jvega190 we need to make sure by fixing this case we're not regressing on a prior fix or case.

**Comment by rart** [comment]
This seems like a dup of https://github.com/craftercms/craftercms/issues/6368.

**Comment by phuongnq** [comment]
Verified fixed in 4.1.3!

**Comment by phuongnq** [comment]
verified fixed in 4.2.0!
